sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -70,8 +75,18 @@ export class CallArgSerializer {
 
   async newObjectArg(objectId: string): Promise<ObjectArg> {
     const object = await this.provider.getObject(objectId);
-    if (isSharedObject(object)) {
-      return { Shared: objectId };
+    const isNewSharedObjectAPI =
+      (await this.provider.getRpcApiVersion()) === 'latest';
+
+    if (!isNewSharedObjectAPI) {
+      if (isSharedObject(object)) {
+        return { Shared_Deprecated: objectId };
+      }
+    } else {
+      const initialSharedVersion = getSharedObjectInitialVersion(object);
+      if (initialSharedVersion) {
+        return { Shared: { objectId, initialSharedVersion } };
+      }
     }
```

Comment:
```suggestion
    const version = await this.provider.getRpcApiVersion();
    const useOldSharedObjectAPI =
      version === '0.11.0'  || version === '0.12.0';

    if (useOldSharedObjectAPI) {
      if (isSharedObject(object)) {
        return { Shared_Deprecated: objectId };
      }
    } else {
      const initialSharedVersion = getSharedObjectInitialVersion(object);
      if (initialSharedVersion) {
        return { Shared: { objectId, initialSharedVersion } };
      }
    }
```
`getRpcApiVersion` will always return a valid version number instead of 'latest' in the future